contracts/src/v0.8/llo-feeds/RewardManager.sol

```diff
@@ -0,0 +1,311 @@
+  bytes32[] public s_registeredPoolIds;
+
+  // @dev The address for the LINK contract
+  address private immutable i_linkAddress;
+
+  // The total weight of all RewardRecipients. 1e18 = 100% of the pool fees
+  uint256 private constant PERCENTAGE_SCALAR = 1e18;
+
+  // The fee manager address
+  address private s_feeManagerAddress;
+
+  // @notice Thrown whenever the RewardRecipient weights are invalid
+  error InvalidWeights();
+
+  // @notice Thrown when any given address is invalid
+  error InvalidAddress();
+
+  // @notice Thrown when the pool id is invalid
+  error InvalidPoolId();
+
+  // @notice Thrown when the calling contract is not within the authorized contracts
+  error Unauthorized();
+
+  // Events emitted upon state change
+  event RewardRecipientsUpdated(bytes32 indexed poolId, Common.AddressAndWeight[] newRewardRecipients);
+  event RewardsClaimed(bytes32 indexed poolId, address indexed recipient, uint256 quantity);
+  event FeeManagerUpdated(address newFeeManagerAddress);
+  event FeePaid(bytes32 poolId, address payee, uint256 quantity);
+
+  /**
+   * @notice Constructor
+   * @param linkAddress address of the wrapped LINK token
+   */
+  constructor(address linkAddress) ConfirmedOwner(msg.sender) {
+    //ensure that the address ia not zero
+    if (linkAddress == address(0)) revert InvalidAddress();
+
+    i_linkAddress = linkAddress;
+  }
+
+  // @inheritdoc TypeAndVersionInterface
+  function typeAndVersion() external pure override returns (string memory) {
+    return "RewardManager 0.0.1";
+  }
+
+  // @inheritdoc IERC165
+  function supportsInterface(bytes4 interfaceId) external pure override returns (bool) {
+    return interfaceId == this.onFeePaid.selector;
+  }
+
+  modifier onlyOwnerOrFeeManager() {
+    if (msg.sender != owner() && msg.sender != s_feeManagerAddress) revert Unauthorized();
+    _;
+  }
+
+  modifier onlyOwnerOrRecipientInPool(bytes32 poolId) {
+    if (msg.sender != owner() && s_rewardRecipientWeights[poolId][msg.sender] == 0) revert Unauthorized();
+    _;
+  }
+
+  /// @inheritdoc IRewardManager
+  function onFeePaid(bytes32 poolId, address payee, Common.Asset calldata fee) external override {
+    //this pool only handles LINK fees
+    if (fee.assetAddress != i_linkAddress) {
+      revert InvalidAddress();
+    }
+
+    //update the total fees collected for this pot
+    unchecked {
+      //the total amount for any ERC20 asset cannot exceed 2^256 - 1
+      s_totalRewardRecipientFees[poolId] += fee.amount;
+    }
+
+    //transfer the fee to this contract
+    IERC20(i_linkAddress).transferFrom(payee, address(this), fee.amount);
+
+    emit FeePaid(poolId, payee, fee.amount);
+  }
+
+  /// @inheritdoc IRewardManager
+  function claimRewards(bytes32[] memory poolIds) external override {
+    _claimRewards(msg.sender, poolIds);
+  }
+
+  // wrapper impl for claimRewards
+  function _claimRewards(address recipient, bytes32[] memory poolIds) internal returns (uint256) {
```

Comment:
Should `poolIds[]` be unique?

Reply:
They should, but it doesn't matter if theirs duplicates, as it will be ultimately the same as calling claimRewards twice, which is handled